inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -142,13 +142,26 @@ InferenceEngine::Parameter MultiDeviceInferencePlugin::GetMetric(const std::stri
     }
 }
 
+// Is called only when caching is enabled
+    InferenceEngine::IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetwork(const std::string& modelPath,
+                                                                           const std::map<std::string, std::string>& config) {
+    CNNNetwork network;
+    return LoadExeNetworkImpl(modelPath, network, config);
```

Comment:
I've just realized that this:
https://github.com/openvinotoolkit/openvino/blob/9a1f4b6829fdb02a7e06ce5b4312e33ce52aee26/inference-engine/src/plugin_api/cpp_interfaces/impl/ie_plugin_internal.hpp#L68-L70
Is not a part of ExecutableNetwork created from file name.. Can we fix this?
1. MultiDeviceExecutableNetwork can take information about inputs and outputs from device `ExecutableNetwork`
2. Make a call `<returned multi exec network>->SetPointerToPlugin(shared_from_this())` (probably, we can create `LoadNetworkImpl` as for `LoadNetwork(CNNNetwork)`, but maybe it's overkill just for a additional single line of code)

Reply:
Well, actually there is (in both Multi & other plugins)
```
GetCore()->LoadNetwork(cnnNet, ...);
````
And this 'LoadNetwork' will go through ICore->Plugin->LoadNetwork(CNNNet) and it will call necessary setNetworkInputs, setNetworkOutputs, SetPointerToPlugin...
I can add appropriate EXPECT_CALL in my caching tests that these methods are called during 'LoadNetwork'

Reply:
thanks!